src/main/java/com/influxdb/v3/client/internal/FlightSqlClient.java

```diff
@@ -130,11 +152,59 @@ public void close() throws Exception {
     private FlightClient createFlightClient(@Nonnull final ClientConfig config) {
         Location location = createLocation(config);
 
-        return FlightClient.builder()
-                .location(location)
-                .allocator(new RootAllocator(Long.MAX_VALUE))
-                .verifyServer(!config.getDisableServerCertificateValidation())
-                .build();
+        final NettyChannelBuilder nettyChannelBuilder = NettyChannelBuilder.forTarget(location.getUri().getHost());
+        if (!validSchemas.contains(location.getUri().getScheme())) {
+            throw new IllegalArgumentException(
+                    "Scheme is not supported: " + location.getUri().getScheme());
+        }
+
+        if (location.getUri().getScheme().equals(LocationSchemes.GRPC_DOMAIN_SOCKET)) {
+            setChannelTypeAndEventLoop(nettyChannelBuilder);
+        }
+
+        if (LocationSchemes.GRPC_TLS.equals(location.getUri().getScheme())) {
+            nettyChannelBuilder.useTransportSecurity();
+
+            SslContext nettySslContext = createNettySslContext(config);
+            nettyChannelBuilder.sslContext(nettySslContext);
+        } else {
+            nettyChannelBuilder.usePlaintext();
+        }
+
+        if (config.getProxyUrl() != null) {
+            ProxyDetector proxyDetector = createProxyDetector(config.getHost(), config.getProxyUrl());
+            nettyChannelBuilder.proxyDetector(proxyDetector);
+        }
+
+        if (config.getProxy() != null) {
+            LOG.warn("proxy property in ClientConfig will not work in query api, use proxyUrl property instead");
+        }
+
+        nettyChannelBuilder.maxTraceEvents(0)
+                .maxInboundMessageSize(Integer.MAX_VALUE)
+                .maxInboundMetadataSize(Integer.MAX_VALUE);
+
+        return FlightGrpcUtils.createFlightClient(new RootAllocator(Long.MAX_VALUE), nettyChannelBuilder.build());
+    }
+
+    @Nonnull
+    SslContext createNettySslContext(@Nonnull final ClientConfig config) {
+        try {
+            SslContextBuilder sslContextBuilder;
+            sslContextBuilder = GrpcSslContexts.forClient();
+            if (!config.getDisableServerCertificateValidation()) {
+                if (config.sslRootsFilePath() != null) {
+                    try (FileInputStream fileInputStream = new FileInputStream(config.sslRootsFilePath())) {
+                        sslContextBuilder.trustManager(fileInputStream);
+                    }
+                }
+            } else {
+                sslContextBuilder.trustManager(InsecureTrustManagerFactory.INSTANCE);
+            }
```

Comment:
In my previous review I missed this detail. If `sslRootsFilePath` is not provided that does not mean that we are disabling the server verification but we are using operating system's default trust store for verification.
```suggestion
            if (config.getDisableServerCertificateValidation()) {
                sslContextBuilder.trustManager(InsecureTrustManagerFactory.INSTANCE);
            } else if (config.sslRootsFilePath() != null) {
                try (FileInputStream fileInputStream = new FileInputStream(config.sslRootsFilePath())) {
                    sslContextBuilder.trustManager(fileInputStream);
                }
            }
```